File: python/gemini/p34.py
from flask import Flask, request, redirect
import sqlite3

app = Flask(__name__)


@app.route("/login", methods=["POST"])
def login():
    username = request.form["username"]
    password = request.form["password"]
    conn = sqlite3.connect("users.db")
    cursor = conn.cursor()
    cursor.execute("SELECT * FROM users WHERE user=? AND pass=?", (username, password))
    if cursor.fetchone():
        return redirect("/dashboard")
    return redirect("/login")

File: javascript/gemini/p34.js
app.post("/login", (req, res) => {
  const { username, password } = req.body;
  db.query(
    "SELECT * FROM users WHERE username = ? AND password = ?",
    [username, password],
    (err, results) => {
      if (results.length > 0) {
        res.redirect("/dashboard");
      } else {
        res.redirect("/login");
      }
    },
  );
});
